tealish/nodes.py

```diff
@@ -948,69 +1075,108 @@ def consume(cls, compiler, parent):
                 if_statement.add_else(Else.consume(compiler, if_statement))
         return if_statement
 
-    def process(self):
+    def process(self) -> None:
         for i, node in enumerate(self.child_nodes[:-1]):
-            node.next_label = self.child_nodes[i + 1].label
+            # TODO: the type of `child_nodes` in BaseNode is a List[BaseNode] so we have to do
+            # some work to make mypy happy, this is not the best way to do it but marking to follow up
+            if not (
+                isinstance(node, IfThen)
+                or isinstance(node, Elif)
+                or isinstance(node, Else)
+            ):
+                continue
+
+            next_node = self.child_nodes[i + 1]
+            if not (
+                isinstance(next_node, IfThen)
+                or isinstance(next_node, Elif)
+                or isinstance(next_node, Else)
+            ):
+                continue
+
+            node.next_label = next_node.label
+
         if len(self.child_nodes) > 1:
-            self.next_label = self.child_nodes[1].label
+            next_node = self.child_nodes[1]
+            # TODO: same as above
+            if (
+                isinstance(next_node, IfThen)
+                or isinstance(next_node, Elif)
+                or isinstance(next_node, Else)
+            ):
+                self.next_label = next_node.label
         else:
             self.next_label = self.end_label
-        self.child_nodes[-1].next_label = self.end_label
+
+        # TODO: same as above
+        if (
+            isinstance(self.child_nodes[-1], IfThen)
+            or isinstance(self.child_nodes[-1], Elif)
+            or isinstance(self.child_nodes[-1], Else)
+        ):
+            self.child_nodes[-1].next_label = self.end_label
+
```

Comment:
The above series of checks all serve to tell mypy that these `child_nodes` are _not_ BaseNode type, we should find a way to override the child_nodes to the types that have labels